Remotion/Web/Development.WebTesting/Accessibility/Implementation/AccessibilityResultMapper.cs

```diff
@@ -0,0 +1,135 @@
+          violations: ParseViolations (axeResult).ToList());
+    }
+
+    private DateTime ParseTimestamp (string timestampAsString)
+    {
+      return DateTime.ParseExact (
+          timestampAsString,
+          "yyyy-MM-ddTHH:mm:ss.fffZ",
+          CultureInfo.CreateSpecificCulture ("en-US"),
+          DateTimeStyles.AssumeLocal);
+    }
+
+    private AccessibilityConformanceLevel ParseConformanceLevel (string[] tags)
+    {
+      var conformanceLevels = tags.Where (AccessibilityConformanceLevelConverter.IsValid);
+      var highestConformanceLevel = conformanceLevels.OrderByDescending (s => s.Length).First();
+      return AccessibilityConformanceLevelConverter.ConvertToEnum (highestConformanceLevel);
+    }
+
+    private IEnumerable<AccessibilityRuleResult> ParseViolations (AxeResult deserializedResult)
+    {
+      foreach (var axeRule in deserializedResult.Violations)
+      {
+        var accessibilityRule = GetAccessibilityRule (axeRule);
+        foreach (var node in axeRule.Nodes)
+        {
+          yield return GetViolation (node, accessibilityRule);
+        }
+      }
+    }
+
+    private AccessibilityRule GetAccessibilityRule (AxeRuleResult ruleResult)
+    {
+      return new AccessibilityRule (
+          AccessibilityRuleIDConverter.ConvertToEnum (ruleResult.ID),
+          ruleResult.Description,
+          ToEnum<AccessibilityTestImpact> (ruleResult.Impact),
+          GetSuccessCriteria (ruleResult.Tags).ToArray());
+    }
+
+    private AccessibilityRuleResult GetViolation (AxeHtmlElementResult htmlElementResult, AccessibilityRule accessibilityRule)
+    {
+      var anyRequirements = GetAccessibilityRequirements (htmlElementResult.Any).ToArray();
+      var allRequirements = GetAccessibilityRequirements (htmlElementResult.All).ToArray();
+      var noneRequirements = GetAccessibilityRequirements (htmlElementResult.None).ToArray();
+
+      if (htmlElementResult.XPaths.Length != htmlElementResult.Target.Length)
+        throw new InvalidOperationException ("The XPath and CSS selectors of the target do not match.");
+
+      var targets = htmlElementResult.XPaths.Zip (htmlElementResult.Target, (xpath, css) => new AccessibilityTestTarget (xpath, css)).ToArray();
+
+      return new AccessibilityRuleResult (accessibilityRule, targets, htmlElementResult.Html, allRequirements, anyRequirements, noneRequirements);
+    }
+
+    private IEnumerable<AccessibilityTestSuccessCriteria> GetSuccessCriteria (string[] ruleTags)
+    {
+      return ruleTags
+          .Where (t => !AccessibilityConformanceLevelConverter.IsValid (t))
+          .Select (ParseAccessibilityTestSuccessCriteria);
+    }
+
+    private AccessibilityTestSuccessCriteria ParseAccessibilityTestSuccessCriteria (string rawTag)
+    {
+      var tagWithReplacedWCAG = Regex.Replace (rawTag, @"^wcag(\d)(\d)(\d{1,2})$", "wcag_$1_$2_$3");
+      var tagWithReplacedWCAGAndSeparators = Regex.Replace (tagWithReplacedWCAG, @"(\.|-)", "_");
```

Comment:
Testcase missing?
Also, I thought that this one is just with section508? Should the regex be more specific?

Reply:
I verified that the second regex is indeed also for aXe stuff like `cat.structure` and not only for section508. Also, I adapted a unit test to also map one of those tags